### PR TITLE
[ACR] Improve login/access authorization to registries

### PR DESCRIPTION
**Pull request by djyou**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py [hunk 1]
```diff
@@ -233,54 +229,50 @@ def acr_login(registry_name, resource_group_name=None, username=None, password=N
...
+        resource_group_name=resource_group_name,
+        username=username,
+        password=password)
+
+    use_password_stdin = False
+    try:
+        p = Popen(["docker", "login", "--help"], stdout=PIPE, stderr=PIPE)
+        stdout, _ = p.communicate()
+        if b'--password-stdin' in stdout:
+            use_password_stdin = True
```
Comment (troydai) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py]: Is there other approach to determine if the underlying docker client support this parameter? For example, can you determine the version of the docker client? Parsing the help doc may not be the most stable or elegant approach. It also worth consider if there is localized docker client.

Comment (djyou) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py]: It wasn't clear to us which version of `docker` has support for `--password-stdin`, besides `docker` has multiple editions. It is indeed not an elegant approach, but enumerating all versions/editions seem to be less stable than parsing the help message. I am not aware of localized docker client.

Comment (troydai) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py]: ok.

**Comment by troydai** [comment]
I will merge once CI passes.

**Pull request merged**